src/main/java/org/springframework/data/elasticsearch/core/SearchHitSupport.java

```diff
@@ -81,4 +88,55 @@ public static Object unwrapSearchHits(Object result) {
 
 		return result;
 	}
+
+	public static <T> SearchPage<T> searchPageFor(SearchHits<T> searchHits, @Nullable Pageable pageable) {
+		return new SearchPageImpl<>(searchHits, pageableOrUnpaged(pageable));
+	}
+
+	public static class SearchPageImpl<T> extends PageImpl<SearchHit<T>> implements SearchPage<T> {
+
+		private final SearchHits<T> searchHits;
+		private Aggregations aggregations;
+
+		public SearchPageImpl(SearchHits<T> searchHits, Pageable pageable) {
+			super(searchHits.getSearchHits(), pageable, searchHits.getTotalHits());
+			this.searchHits = searchHits;
+		}
+
+		@Override
+		public SearchHits<T> getSearchHits() {
+			return searchHits;
+		}
+
+		@Override
+		public boolean hasAggregations() {
+			return aggregations != null;
+		}
+
+		@Override
+		public Aggregations getAggregations() {
+			return aggregations;
+		}
+
+		@Override
+		public Aggregation getAggregation(String name) {
+			return aggregations == null ? null : aggregations.get(name);
+		}
+
+		@Override
+		public float getMaxScore() {
+			return searchHits.getMaxScore();
+		}
+
+		@Override
+		@Nullable
+		public String getScrollId() {
+			return searchHits.getScrollId();
+		}
+	}
+
+	private static Pageable pageableOrUnpaged(Pageable pageable) {
+		return ofNullable(pageable).orElse(Pageable.unpaged());
```

Comment:
Why not a simple null check?

Reply:
I took this from `AggregatePageImpl` because it's a nice one-liner. The null check version would be:
```
	private static Pageable pageableOrUnpaged(Pageable pageable) {
		if(pageable != null) {
			return pageable;
		} else {
			return Pageable.unpaged();
		}
        }
```